libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObjectDistance.kt

```diff
@@ -0,0 +1,53 @@
+package com.mapbox.navigation.base.trip.model.roadobject
+
+/**
+ * RoadObjectDistance contains information about road object id, type and information about distance
+ * to the road object of concrete flavor/shape.
+ *
+ * @param roadObjectId id of the road object
+ * @param type type of the road object
+ * @param distanceInfo extra distance info
+ */
+class RoadObjectDistance(
+    val roadObjectId: String,
+    @RoadObjectType.Type val type: Int,
+    val distanceInfo: RoadObjectDistanceInfo,
+) {
+
```

Comment:
```suggestion
class RoadObjectDistance(
    val roadObjectId: String,
    @RoadObjectType.Type val type: Int,
    @RoadObjectDistanceInfoType.Type val distanceInfoType: Int
) {
/**
 * Distance to start of the object from current location
 */
abstract val distanceToStart: Double
  
```

Why `RoadObjectDistanceInfo` have to be a separate object?

Reply:
because it's a base class for `LineDistanceInfo`, `PolygonDistanceInfo`, etc.

Reply:
do you have any concerns if we merge models?
cc @etl @SiarheiFedartsou

Reply:
No, I don't, but let's wait for Sergey's opinion.

Reply:
Looks good to me.